File: C++/Metodos/Metodo.cpp
#include<iostream>
using namespace std;

class Barcos{
		public:
		void prenderBarco(string marca, string modelo, string color, double precio) {//metodo vacio que posee cuatro parametros el cual se dara desde el metodo main
        cout<<"* El barco de marca "<<marca<<" de modelo "<<modelo<< " con el color "<<color<< " posee un precio de $"<<precio<<" este se esta prendiendo."<<endl;//concateno la variable con el valor dado
    }

    void arrancarBarco(string marca, string modelo, string color, int precio){
       cout<<"* El barco de marca "<<marca<<" de modelo "<<modelo<< " con el color "<<color<< " posee un precio de $"<<precio<<" este se esta arrancando."<<endl;
    }
    
    void frenarBarco (string marca, string modelo, string color, int precio){
    
        cout<<"* El barco de marca "<<marca<<" de modelo "<<modelo<< " con el color "<<color<<" posee un precio de $"<<precio<<" este se esta frenando."<<endl;
    }
    
    void acelerarBarco(string marca, string modelo, string color, int precio){
        cout<<"* El barco de marca "<<marca<<" de modelo "<<modelo<< " con el color "<<color<< " posee un precio de $"<<precio<<" este se esta acelaerando."<<endl;
    }

    void apagarBarco(string marca, string modelo, string color, int precio){
        cout<<"* El barco de marca "<<marca<<" de modelo "<<modelo<< " con el color "<<color<< " posee un precio de $"<<precio<<" este se esta apagando."<<endl;
    }
	};
int main(){
     	//aqui crearemos nuestras instancias o objetos  
        cout <<"Bienvenido..!"<< endl;
        cout <<" "<< endl;
        cout <<"----------------------------------------"<< endl;
        cout <<"Primer Barco: "<< endl;
        //creacion de los constructores
        Barcos primerBarco = Barcos();
        primerBarco.prenderBarco("Nautic","2017", "Rojo", 1500000);
        primerBarco.arrancarBarco("Ninco", "2005", "blanco", 350000);
        primerBarco.acelerarBarco("Honda", "2003", "dorado", 220000);
        primerBarco.frenarBarco("spice", "2001", "verde", 1800000);//cosntructor llamando a los metodos y pasando los parametros
        primerBarco.apagarBarco("old", "2006", "plata", 12500000);//de la misma manera que en java
        cout <<"----------------------------------------"<< endl;
        cout <<" "<< endl;
        cout <<"Segundo Barco:"<< endl;
        
        Barcos segundoBarco =Barcos();//segundo conscructor
        segundoBarco.prenderBarco("spice", "2003", "negro", 0);
        segundoBarco.arrancarBarco("BMW", "2017", "Verde", 4500000);
        segundoBarco.acelerarBarco("nautic", "2008", "naranja", 820000);
        segundoBarco.frenarBarco("capelli", "2009", "azul", 650400);
        segundoBarco.apagarBarco("geleon", "2016", "azulado", 785000);
        cout <<"----------------------------------------"<< endl;
        cout <<" "<< endl;
        cout <<"Tercer Barco: "<< endl;
        
        Barcos tercerBarco = Barcos();
        tercerBarco.prenderBarco("Le boat","2008","",1540000);
        tercerBarco.arrancarBarco("ford","2013","blanco",189000);//parametros utilizados para cada metodo
        tercerBarco.acelerarBarco("Le boat", "2010","gris", 380000);
        tercerBarco.frenarBarco("nautic","2005","cafe",1350020);
        tercerBarco.apagarBarco("valli","1999","negro",4950000);
        cout <<"----------------------------------------"<< endl;
        cout <<" "<< endl;
        cout <<"Cuarto Barco: "<< endl;
        
        Barcos cuartoBarco = Barcos();//ultimo constructor
        cuartoBarco.prenderBarco("Astromar","2007","negro",1000000);
        cuartoBarco.arrancarBarco("nautic","2013","blanco",1540000);
        cuartoBarco.acelerarBarco("valli","2005","azul",1900000);
        cuartoBarco.frenarBarco("Astromar", "2015", "negro", 1805000);
        cuartoBarco.apagarBarco("leboat","2016","cafe", 45000000);
        cout <<"----------------------------------------"<< endl;
	return 0;
}

File: Java/Metodos/Metodo.java
/*Programa hecho por Harold Esteban Pedraza tercer programa documentado para fundamento poo, hecho en netbeans el dia 18/10/2017 hora 21:30 pm
 *Este ejercisio tiene como proposito crear acciones de las clases
 *que finalmente puede hacer todos los objetos de la clase
 */
package metodos;

/**Declaracion de la clase Metodo
 * @author Harold Pedraza
 * @version 18/10/2017
 */

public class Metodo {

    public static void main(String[] args) {
        /*
        Crear objetos para usar los metodos que estan en la clase barcos
        Primeramente para poder utilizar los atributos y metodos en conjunto, en este caso debemos crear variables de atributos de los barcos
        en este caso color, marca, precio y modelo y cada uno de ellas cambia los valores independientemente de cuantos barcos requiera
        */
        String color1= "Blanco", color2 ="negro", color3 = "azul", color4 = "verde";
        String marca1= "nautic", marca2 = "nau", marca3 = "blue", marca4 = "stack";
        int precio1 = 10000, precio2 = 15000, precio3 = 75000, precio4 = 80000;//ATRIBUTOS DE LOS BARCOS CREADOS
        int modelo1 = 2017, modelo2 = 2010, modelo3 = 2007, modelo4 = 2000;
        /*
        Con las variables ya creadas eh inicializadas, solo falta agrupar cada una de estas a el barco que deseemos
        Luego creamos constructores en este caso necesito los atributos de cuatro barcos entonces necesito igualmente cuatro constructores
        que me premitan abrir lo creado en la clase Barcos
        */
        Barcos primerBarco = new Barcos();
        System.out.println("Bienvenido...!");
        System.out.println("-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-");
        System.out.println("Primer Barco = ");
        System.out.println(" ");
        System.out.print("El barco de marca ");
        primerBarco.marca(marca1);//En esta parte ya que en las funciones creadas en la otra clase coloque parametro utilizo eso para que ese paramentro sea mi variables creada
        System.out.print("De color ");
        primerBarco.color(color1);
        System.out.print("Con el precio de ");
        primerBarco.precio(precio1);//De igual manera paso el parametro de la variable inicializada a la funcion correspondiente
        System.out.print("De modelo ");
        primerBarco.modelo(modelo1);
        System.out.println("         HACE ");
        /*
        En esta parte utilizo el constructor para abrir las funciones y ejecutar lo que tiene cada uno de ellas.
        */
        primerBarco.prenderBarco();
        primerBarco.arrancarBarco();
        primerBarco.acelerarBarco();
        primerBarco.frenarBarco();
        primerBarco.apagarBarco();
        
        Barcos segundoBarco = new Barcos();
        /*
        De la misma manera que el anterior solo que este sera mi segundo barco y ya que es el segundo necessito diferentes precios
        marcas, color entonces lo diferencio del anterior con numeros 1,2,3,4, igualmente creo el constructor y llamo funcion por
        funcion y a los atributos les paso el paramentro de mi variables creada para que see ejecute bien el metodo
        */
        System.out.println(" ");
        System.out.println("-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-");
        System.out.println("Segundo Barco = ");
        System.out.println();
        System.out.print("El barco de marca ");
        segundoBarco.marca(marca2);
        System.out.print("De color ");
        segundoBarco.color(color2);
        System.out.print("Con el precio de ");
        primerBarco.precio(precio2);
        System.out.print("De modelo ");
        primerBarco.modelo(modelo2);
        System.out.println("         HACE ");
        segundoBarco.prenderBarco();
        segundoBarco.arrancarBarco();
        segundoBarco.acelerarBarco();
        segundoBarco.frenarBarco();
        segundoBarco.apagarBarco();
        
        Barcos tercerBarco = new Barcos();//De igual manera tercer constructor creado con diferentes variables inicializadas
        System.out.println(" ");
        System.out.println("-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-");
        System.out.println("Tercer Barco = ");
        System.out.println();
        System.out.print("El barco de marca ");
        tercerBarco.marca(marca3);
        System.out.print("De color ");
        tercerBarco.color(color3);
        System.out.print("Con el precio de ");
        primerBarco.precio(precio3);
        System.out.print("De modelo ");
        primerBarco.modelo(modelo3);
        System.out.println("         HACE");
        tercerBarco.prenderBarco();
        tercerBarco.arrancarBarco();
        tercerBarco.acelerarBarco();
        tercerBarco.frenarBarco();
        tercerBarco.apagarBarco();
        
        Barcos cuartoBarco = new Barcos();
        System.out.println(" ");
        System.out.println("-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-");
        System.out.println("Cuarto Barco = ");
        System.out.println();
        System.out.print("El barco de marca ");
        tercerBarco.marca(marca4);
        System.out.print("De color ");
        tercerBarco.color(color4); 
        System.out.print("Con el precio de ");
        primerBarco.precio(precio4);
        System.out.print("De modelo ");
        primerBarco.modelo(modelo4);
        System.out.println("         HACE ");
        cuartoBarco.prenderBarco();
        cuartoBarco.arrancarBarco();
        cuartoBarco.acelerarBarco();
        cuartoBarco.frenarBarco();
        cuartoBarco.apagarBarco();
        System.out.println("-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-/-");
    }  
}
